### PR TITLE
including the photon table

### PR DESCRIPTION
**Pull request by amaringarcia**

### CODE REVIEW

File: ALICE3/DataModel/PHOTON.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);   //! Index to travel from PHOTON to collision
+DECLARE_SOA_INDEX_COLUMN(Track, track);           //! Index to travel from PHOTON to track
+DECLARE_SOA_INDEX_COLUMN(McParticle, mcparticle); //! Index to travel from PHOTON to particle
+DECLARE_SOA_COLUMN(PX, px, float);                //! Px
+DECLARE_SOA_COLUMN(PY, py, float);                //! Py
+DECLARE_SOA_COLUMN(PZ, pz, float);                //! Pz
+} // namespace alice3photon
+
+DECLARE_SOA_TABLE(PHOTONs, "AOD", "PHOTONCONV", //! Table for the ALICE3 Photon detector
```
Comment (vkucera) [inline ALICE3/DataModel/PHOTON.h:38]: > Type names follow camel case convention

File: ALICE3/DataModel/PRES.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+namespace o2::aod
+{
+namespace alice3pres
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);              //!
+DECLARE_SOA_INDEX_COLUMN(Track, track);                      //!
+DECLARE_SOA_COLUMN(PRESIsElectron, presIsElectron, uint8_t); //! FIXME: To be changed to bool once bool columns are groupable.
+} // namespace alice3pres
+
+DECLARE_SOA_TABLE(PRESs, "AOD", "PRES", //!
```
Comment (vkucera) [inline ALICE3/DataModel/PRES.h:34]: > Type names follow camel case convention

File: ALICE3/Tasks/CMakeLists.txt [hunk 3]
```diff
@@ -43,3 +43,9 @@ o2physics_add_dpl_workflow(alice3-cdeuteron
                     SOURCES alice3-cdeuteron.cxx
                     PUBLIC_LINK_LIBRARIES O2Physics::AnalysisCore O2::DetectorsVertexing
                     COMPONENT_NAME Analysis)
+
+
+o2physics_add_dpl_workflow(alice3-photon-qa
```
Comment (vkucera) [inline ALICE3/Tasks/CMakeLists.txt:48]: Do not add unnecessary blank lines.

File: ALICE3/Tasks/PHOTONqa.cxx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PHOTONqa.cxx
```
Comment (vkucera) [inline ALICE3/Tasks/PHOTONqa.cxx:13]: > File names follow camel case convention

File: ALICE3/Tasks/PHOTONqa.cxx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+DECLARE_SOA_INDEX_COLUMN(McParticle, mcparticle);
+
+} // namespace indices
+
+DECLARE_SOA_INDEX_TABLE_USER(PHOTONTracksIndex, Tracks, "PHOTONTRK", indices::TrackId, indices::PHOTONId);
+DECLARE_SOA_INDEX_TABLE_USER(PHOTONMcPartIndex, McParticles, "PHOTONPART", indices::McParticleId, indices::PHOTONId);
+} // namespace o2::aod
+
+struct photonIndexBuilder { // Builder of the PHOTON-track index linkage
```
Comment (vkucera) [inline ALICE3/Tasks/PHOTONqa.cxx:50]: Structs have the same naming conventions as types.

File: ALICE3/Tasks/PHOTONqa.cxx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  Builds<o2::aod::PHOTONTracksIndex> ind;
+  Builds<o2::aod::PHOTONMcPartIndex> indPart;
+  void init(o2::framework::InitContext&)
```
Comment (vkucera) [inline ALICE3/Tasks/PHOTONqa.cxx:53]: Separate the function with a blank line.

File: ALICE3/Tasks/PHOTONqa.cxx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        continue;
+      const float photonp = std::sqrt(particle.photon().px() * particle.photon().px() + particle.photon().py() * particle.photon().py() + particle.photon().pz() * particle.photon().pz());
+      histos.fill(HIST("photonpVsp"), photonp, particle.p());
+      histos.fill(HIST("photonpxVspx"), particle.photon().px(), particle.px());
+      histos.fill(HIST("photonpyVspy"), particle.photon().py(), particle.py());
+      histos.fill(HIST("photonpzVspz"), particle.photon().pz(), particle.pz());
+      histos.get<TH1>(HIST("PDGs"))->Fill(Form("%i", particle.pdgCode()), 1.f);
+    }
+    for (auto& photon : photons) {
+      const float photonp = std::sqrt(photon.px() * photon.px() + photon.py() * photon.py() + photon.pz() * photon.pz());
```
Comment (vkucera) [inline ALICE3/Tasks/PHOTONqa.cxx:89]: Seems you could improve the readability by defining `pX`, `pY`, `pZ`.
Also using `RecoDecay::sqrtSumOfSquares` might be useful here.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Hi @amaringarcia , any updates on this?

File: ALICE3/DataModel/PHOTON.h (hunk 1 above)
Comment (amaringarcia) [reply ALICE3/DataModel/PHOTON.h:38]: I have included these modifications in the chic update

File: ALICE3/DataModel/PRES.h (hunk 2 above)
Comment (amaringarcia) [reply ALICE3/DataModel/PRES.h:34]: I have delete this file, ad the preshower detector will not be build

**Pull request closed without merging**